Ond232 211 refactor t2s request message

fcavallin opened the pull request:
refactor parameters of synthesize request to separate message

Commit:
Refactored request message

Commit:
Refactored request message

Commit:
audioformat and pcm oneof added and documentation started

Commit:
added uuid

Commit:
Refactored request message, uuid added and batchsynthesize

JSteinbauer commented on ondewo/t2s/text-to-speech.proto:
```diff
@@ -19,34 +19,107 @@ import "google/protobuf/empty.proto";
+    // Optional. This parameter is used for time stretching which is the process of
+    // changing the speed or duration of an audio.
+    // It should be much more than 1.0. O is not a valid number for this variable.
+    // The default value is 1.
```
There seems to be a small problem here: You say that this field is optional, and that its default value is 1. If a field is optional, it's up to you to provide it or not - but if you don't explicitly set it here, it will just take the default value for a grpc proto float which is 0.

JSteinbauer commented on ondewo/t2s/text-to-speech.proto:
```diff
@@ -19,34 +19,107 @@ import "google/protobuf/empty.proto";
+
+    // Optional. Defines the sample rate of the generated wav file.
+    // The default value is 22050.
+    int32 sample_rate = 4;
```
Idk if this default value is always the same? maybe re-formulate "A good default value is 22050" - if that's what you mean.

andrey999333 replied:
no, it's exactly as it's written - default value is 22050, meaning if you put sample_rate=0 it will change to 22050, since 0 is not valid value

JSteinbauer commented on ondewo/t2s/text-to-speech.proto:
```diff
@@ -19,34 +19,107 @@ import "google/protobuf/empty.proto";
+
+    // Optional. Define if cache should be used or not.
+    // The default value is False.
+    bool use_cache = 7;
```
Is the default value always "False"? If not, this could cause problems, since not setting use_cache explicitly is equivalent to setting it to False.

JSteinbauer commented on ondewo/t2s/text-to-speech.proto:
```diff
@@ -55,98 +128,182 @@ message SynthesizeRequest {
-    Postprocessing postprocessing = 6;
+
+    // Required. Defines the description of the pipeline representation.
+    Description description = 3;
```
Nice :1st_place_medal:

JSteinbauer commented on ondewo/t2s/text-to-speech.proto:
```diff
@@ -55,98 +128,182 @@ message SynthesizeRequest {
+// Filter pipelines by attributed in request.
 message ListT2sPipelinesRequest {
+
+    // Optional. Define the language/ languages.
```
:+1:

JSteinbauer commented on ondewo/t2s/text-to-speech.proto:
```diff
@@ -19,34 +19,107 @@ import "google/protobuf/empty.proto";
+    // Optional. Defines the noise in the generated audio.
+    // It should be between 0.0 and 1.
+    // The default value is 0.0.
+    float noise_scale = 3;
```
If, for some reason, the value indicated in the pipeline is not 0.0, you will have the same problem as above (but maybe this is never the case?).

JSteinbauer commented on ondewo/t2s/text-to-speech.proto:
```diff
@@ -19,34 +19,107 @@ import "google/protobuf/empty.proto";
 // endpoints of t2s generate service
 service Text2Speech {
+
+    // Synthesizes an specific text sent in the request with the configuration requirements and retrieves a response
```
Nice! :)

JSteinbauer reviewed (commented):
Nice PR! :+1: I just left a couple of comments regarding default values, it think it would be good to change them before merging..

JSteinbauer commented on ondewo/t2s/text-to-speech.proto:
```diff
@@ -19,34 +19,107 @@ import "google/protobuf/empty.proto";
+    RequestConfig config = 2;
+}
+
+message BatchSynthesizeRequest {
```
I think it's not a good idea to have repeated fields for the text and the configs. What if something goes wrong, and you send more texts than configs (or vice-versa)? Why not just having one single field: 
_repeated SynthesizeRequest requests = 1;_

JSteinbauer commented on ondewo/t2s/text-to-speech.proto:
```diff
@@ -19,34 +19,107 @@ import "google/protobuf/empty.proto";
+    repeated RequestConfig config = 2;
+}
+
+message BatchSynthesizeResponse {
```
Just like here - that looks much better :)

Commit:
Changed tuple for a new created t2sPipeline class
Addressed pull request comments

Commit:
added _replace_default_values_request_config method
added a utils file
changed length_scale, noise_scale, sample_rate and use_cache to oneof

Commit:
corrected variable names

andrey999333 commented on ondewo/t2s/text-to-speech.proto:
```diff
@@ -55,98 +134,191 @@ message SynthesizeRequest {
-    Postprocessing postprocessing = 6;
+
+    // Required. Defines the description of the pipeline representation.
+    Description description = 3;
```
it's not at all important but it's cleaner to have 2 after 1 instead of 3

andrey999333 commented on ondewo/t2s/text-to-speech.proto:
```diff
@@ -222,7 +393,6 @@ message MbMelganTriton {
     string triton_url = 4;
```
here it could be 3 instaed of 4

andrey999333 reviewed (approved):
some minor changes could be done, and than it's can be merged

Commit:
changed numbers in proto file for Text2SPeechConfig and Text2Mel
